docs/rfcs/0044-feature-auto-archive.md

```diff
@@ -0,0 +1,230 @@
+# Summary
+
+We will implement an environment-level auto-archive system that enables administrators to automatically archive unused feature flags across an entire environment. This feature addresses the challenge of managing stale feature flags at scale by providing centralized, environment-wide automated cleanup with safety guarantees through dependency checks and code reference validation.
+
+The auto-archive system will allow administrators to configure at the environment level:
+
+- **Enable/Disable auto-archiving**: Toggle auto-archiving for all flags in the environment
+- **Unused days threshold**: Minimum number of days flags must be unused before archiving
+- **Code reference requirements**: Whether to require zero code references before archiving
+
+## Background
+
+As feature flag systems mature, organizations accumulate hundreds or thousands of flags, many of which become obsolete over time. Managing these flags individually is time-consuming and error-prone, often leading to:
+
+- **Technical debt**: Unused flags cluttering the codebase and dashboard
+- **Maintenance burden**: Developers spending time managing obsolete flags manually
+- **Inconsistent cleanup**: Different retention policies applied to different flags
+- **Performance impact**: Evaluation overhead from unnecessary flags
+
+The existing Bucketeer system provides:
+
+- Manual archive functionality via `ArchiveFeature` API
+- `Feature.archived` field (boolean) for marking archived flags
+- Dependency checking via `HasFeaturesDependsOnTargets`
+- Last used tracking via `feature_last_used_info` table
+- Code reference tracking for flags
+
+This RFC proposes adding environment-level auto-archive configuration, allowing administrators to set a single policy that applies to all flags in an environment, significantly reducing the management overhead.
+
+## Goals
+
+
+- Enable environment-wide automated archiving of unused feature flags
+- Provide centralized configuration at the environment level
+- Maintain safety through dependency validation and code reference checks
+- Provide audit trail for all auto-archive operations
+- Reuse existing archive infrastructure for consistency
+- Simplify management compared to per-flag configuration
+
+
+## Implementation
+
+### Database Schema Changes
+
+Add three columns to the existing `environment_v2` table to store auto-archive configuration:
+
+```sql
+ALTER TABLE environment_v2
+  ADD COLUMN auto_archive_enabled BOOLEAN NOT NULL DEFAULT FALSE,
+  ADD COLUMN auto_archive_unused_days INT NOT NULL DEFAULT 90,
+  ADD COLUMN auto_archive_require_no_code_refs BOOLEAN NOT NULL DEFAULT TRUE;
```

Comment:
As environmental options increase, perhaps we should consider designing a proper structure for this? WDYT?

Reply:
Sounds good!
Let's merge it after adding the decision section.